backend/api/views.py

```diff
@@ -60,59 +60,51 @@ def cancel(self, request, pk=None):
     @action(detail=False, methods=['post'], name='Submit new job')
     def submit(self, request):
         serializer = PrintRequestSerializer(data=request.data)
-        if not serializer.is_valid():
-            return Response(serializer.errors,
-                            status=status.HTTP_400_BAD_REQUEST)
+        serializer.is_valid(raise_exception=True)
         printer_with_perms = Printer.get_printer_for_user(user=self.request.user,
                                                           printer_id=serializer.validated_data['printer'])
         if not printer_with_perms:
-            return Response("Printer does not exist", status=status.HTTP_400_BAD_REQUEST)
+            raise ParseError("Selected printer does not exist")
```

Comment:
You're using `ParseError` in multiple places, this is not the correct class for "entity not found" errors. Use the `NotFound` class instead.